.github/workflows/tag_release.yml

```diff
@@ -18,11 +18,37 @@ jobs:
     - name: Set GOVERSION
       run: |
         echo "::set-env name=GOVERSION::$(go version)"
+    - name: Install Ruby
+      uses: actions/setup-ruby@v1
+      with:
+        ruby-version: '2.7'
+    - name: Install github_changelog_generator
+      run: gem install github_changelog_generator
+    - name: Generate changelogs
+      run: make changelog release-changelog
+      env:
+        CHANGELOG_GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
     - name: Run GoReleaser
       uses: goreleaser/goreleaser-action@v1
       with:
         version: "v0.131.1"
-        args: release --rm-dist
+        args: release --rm-dist --release-notes=RELEASE_CHANGELOG.md
       env:
         GOVERSION: ${{ env.GOVERSION }}
         GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    - name: Set GIT_TAG
+      run: |
+        echo "::set-env name=GIT_TAG::${GITHUB_REF/refs\/tags\//}"
+    - name: Commit CHANGELOG.md
+      env:
+        GITHUB_TOKEN: ${{ secrets.github_token }}
+        COMMIT_MSG: |
+          Release ${GIT_TAG}
+          skip-checks: true
+      run: |
+        git config user.email "patrick@fastly.com"
+        git config user.name "phamann"
+        git remote set-url origin https://x-access-token:${GITHUB_TOKEN}@github.com/${GITHUB_REPOSITORY}.git
+        git checkout master
+        git add .
+        git diff --quiet && git diff --staged --quiet || (git commit -m "${COMMIT_MSG}"; git push origin master)
```

Comment:
The master branch is protected as of this review comment. How do we plan to allow the CHANGELOG.md file to be updated with these settings in place?

Reply:
I've tested commit logic on a test branch with branch protection enabled (https://github.com/fastly/cli/commit/864eb8aefb3ca991acdb0b16d22508a92968a742) and it seemed to work. However, this may have only worked because the token I created could have assumed my admin privileges for the repo. It's also not that clear from the actions docs as to whether the generated `GITHUB_TOKEN` assumes admin privileges: https://help.github.com/en/actions/configuring-and-managing-workflows/authenticating-with-the-github_token#about-the-github_token-secret so we may have to test this in CI to determine the behaviour.

Reply:
Worst-case scenario we just need to fix it. Let's move forward.